projects/core/src/cart/facade/cart.service.ts

```diff
@@ -1,73 +1,79 @@
 import { Injectable } from '@angular/core';
 import { select, Store } from '@ngrx/store';
-import { combineLatest, Observable } from 'rxjs';
-import { filter, take, tap } from 'rxjs/operators';
-import { AuthService, UserToken } from '../../auth/index';
+import { asyncScheduler, combineLatest, Observable } from 'rxjs';
+import { debounceTime, filter, map, shareReplay, tap } from 'rxjs/operators';
+import { AuthService } from '../../auth/index';
 import { Cart } from '../../model/cart.model';
 import { OrderEntry } from '../../model/order.model';
-import { BaseSiteService } from '../../site-context/index';
 import * as fromAction from '../store/actions';
 import { StateWithCart } from '../store/cart-state';
 import * as fromSelector from '../store/selectors';
 import { ANONYMOUS_USERID, CartDataService } from './cart-data.service';
 
 @Injectable()
 export class CartService {
+  private prevCartUserId = null;
+  private _activeCart$;
+
   constructor(
     protected store: Store<StateWithCart>,
     protected cartData: CartDataService,
-    protected authService: AuthService,
-    protected baseSiteService: BaseSiteService
+    protected authService: AuthService
   ) {
-    this.init();
+    this._activeCart$ = combineLatest([
+      this.store.select(fromSelector.getCartContent),
+      this.store.select(fromSelector.getCartLoading),
+      this.authService.getUserToken(),
+    ]).pipe(
+      // combineLatest emits multiple times on each property update instead of one emit
+      // additionally dispatching actions that changes selectors used here needs to happen in order
+      // for this asyncScheduler is used here
+      debounceTime(1, asyncScheduler),
+      tap(([cart, loading, userToken]) => {
+        if (
+          // we are only interested in case when we switch from not logged user to logged
+          // - check for userToken.userId !== undefined is used to ignore this action on logout
+          // - check for this.prevCartUserId !== null is used to ignore this action on page refresh for logged in user
+          this.prevCartUserId !== userToken.userId &&
+          typeof userToken.userId !== 'undefined' &&
+          this.prevCartUserId !== null &&
+          !loading
+        ) {
+          this.loadOrMerge();
+        }
+        if (!loading && this.isCreated(cart) && this.isIncomplete(cart)) {
+          this.loadDetails();
+        }
+        this.prevCartUserId = userToken.userId;
+      }),
+      filter(
+        ([cart, loading]) =>
+          !loading &&
+          ((this.isCreated(cart) && !this.isIncomplete(cart)) ||
```

Comment:
It filter out the incomplete cart. So, maybe this filter can be 
`=>!loading && this.isInComplete(cart)`

Reply:
It *filters out* incomplete cart only when it's created
See: `this.isCreated(cart) && !this.isIncomplete(cart)`

When it's not created i.e.`{guid:undefined, code:undefined`, we want to emit this value. It helps to render 'Your shopping cart is empty' in the cart page.